### PR TITLE
Bot accounts UI

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the UI for project bot accounts (listing, removing, adding).

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

image

image

image

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com` or any other user
2. Create a project
3. From the sidebar on the project show page, click `Settings` and then `Bot Accounts`
4. You should not see any bot accounts
5. Click the `New bot account` button
6. Fill in the token name, access level, option expiration date for token, and select atleast 1 scope
7. Click `Submit`
8. You should see a table with the bot account that was created and above the table a section which lists the personal access token
9. Go to the `Members` page for the project
10. You should see the bot account that was created listed as a `Bot`
11. Go back to the `Bot Accounts` page for the project
12. Click the ellipsis for the bot account and click `Remove`. Confirm that you would like to remove the bot account
13. The table should update and no bot accounts should be listed
14. Go back to the `Members` page for the project
15. Verify that the bot account has also been removed from the project members

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/views/projects/bots/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<%= render Viral::PageHeaderComponent.new(title: t('.title'), subtitle: t('.subtitle')) do |component| %>
+  <%= component.with_icon(name: "users", classes: "h-14 w-14 text-primary-700") %>
+  <%= component.with_buttons do %>
+    <%= link_to t(".add_new_bot"),
+    new_namespace_project_bot_path(@namespace.parent, @namespace.project),
+    data: {
+      turbo_frame: "bot_modal",
+      turbo_stream: true
+    },
+    class: "button button--state-primary button--size-default" %>
```
Comment (joshsadam) [inline app/views/projects/bots/index.html.erb:10]: **nitpick:** Can you update these buttons too

**Review by joshsadam — CHANGES_REQUESTED** [review]
Just a few style things.  Everything seems to work as advertised.  Great work.

File: app/views/projects/bots/_bot_modal.html.erb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        <% end %>
+        </div>
+
+      </div>
+        <div class="my-4">
+          <%= form.submit t('projects.bots.index.bot_listing.new_bot_modal.submit'),
+                      class: "button button--size-default button--state-primary" %>
+          <button
+            type="button"
+            class="button button--size-default button--state-default"
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply app/views/projects/bots/_bot_modal.html.erb]: Updated in 93d4124ba618febbc1536fc496f62a267f742847

File: app/views/projects/bots/index.html.erb (hunk 1 above)
Comment (deepsidhu85) [reply app/views/projects/bots/index.html.erb]: Updated in 93d4124ba618febbc1536fc496f62a267f742847

File: app/views/projects/bots/_bot_modal.html.erb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                id="<%= "bot_scopes_#{scope}_help-text" %>"
+                class="text-xs font-normal text-slate-500 dark:text-slate-300"
+              ><%= t("auth.scopes.#{scope}") %></p>
+            </div>
+          </div>
+        <% end %>
+        </div>
+
+      </div>
+        <div class="my-4">
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply app/views/projects/bots/_bot_modal.html.erb]: Fixed in 1d5467da9de9b7f09b1d0b6d20ecbe40e7374469

**Review by joshsadam — APPROVED** [review]
Thanks for the changes.  This looks great and appears to function as it is supposed to.

**Review by joshsadam — APPROVED** [review]
Thanks for the changes.  This looks great and appears to function as it is supposed to.

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**